src/plugins/array.js

```diff
@@ -30,7 +31,9 @@ module.exports = {
     if (mods.remove && !Array.isArray(mods.remove)) throw new Error(`Remove property must be an array`)
     mods.add = mods.add || []
     mods.remove = mods.remove || []
+    // Deep assign nested objects
+    const assign = (arr) => arr.map((x) => typeof x === 'object' ? deepAssign({}, x) : x)
     // Apply modifications and return
-    return without(mods.remove, [ ...data ]).concat([ ...mods.add ])
```

Comment:
I think what you want is:

```js
without(mods.remove, [...data.map(assign)].concat([...mods.add.map(assign)])
```

You're already shallow-cloning the array with `...[]`, so unless `deep-assign` maps over all array elements and deep clones them, that's what you'll need to do yourself.